src/Relude/Bool/Guard.hs

```diff
@@ -78,6 +79,16 @@ guardM :: MonadPlus m => m Bool -> m ()
 guardM f = f >>= guard
 {-# INLINE guardM #-}
 
+-- | 'guarded'. Either lifts a value into an alternative context or gives a minimal value depending on a predicate.
+--
+-- @
+-- guarded even 3 :: [Int] -- []
+-- guarded even 2 :: [Int] -- [2]
+-- @
```

Comment:
We're using `doctest` in this project -- it's a Haskell testing library that automatically tests documentation examples if you write them in a special syntax. For these tests to trigger `doctest` check you need to write them like this:

```haskell
-- >>> guarded even 3 :: [Int]
-- []
```

Also, could I ask you to use bloc style Haddock comments with `{- |` and `-}`? They should be used according to our style guide (because they ere more convenient) but historically in this module the single-line style was used (which I'm going to fix by myself, but I would appreciate if you rewrite this function in this new style). You can find an example here:

https://github.com/kowainik/relude/blob/822672f8901fdfdbb07c956ff50cdfc5eb6be5c5/src/Relude/Monad/Either.hs#L75-L81

Reply:
I am happy to go through and change all the single line comments to multi line. If that would make this PR too large I can open up another issue.